tests/memex/storage_test.py

```diff
@@ -407,10 +434,10 @@ def fetch_annotation(patch):
 
 
 @pytest.fixture
-def models(patch):
-    models = patch('memex.storage.models', autospec=False)
-    models.Annotation.return_value.is_reply = False
-    return models
+def patched_annotation(patch):
+    klass = patch('memex.storage.models.Annotation')
+    klass.return_value.is_reply = False
+    return klass
```

Comment:
I think that what was happening, that was causing `assert_called_once_with('anything')` to always return a truthy mock object and never raise, was that `memex.storage.models` was first being patched and replaced with a mock, and _then_ the `update_document_metadata` fixture was being called and patching `'memex.storage.models.update_document_metadata'`, which was _already_ a mock, so you get a mock of a mock which breaks mock's `assert_called_once_with()`.